prymer/primer3/primer3.py

```diff
@@ -244,8 +253,20 @@ def __init__(
             executable="primer3_core" if executable is None else executable
         )
         command: list[str] = [f"{executable_path}"]
+        if vcfs_or_lookup is None or isinstance(vcfs_or_lookup, VariantLookup):
+            self.vcfs_or_lookup = vcfs_or_lookup
+        elif isinstance(vcfs_or_lookup, list) and all(
+            isinstance(path, Path) for path in vcfs_or_lookup
+        ):  # if provided path, create appropriate `VariantLookup`
+            self.vcfs_or_lookup = VariantLookup.build(
+                vcf_paths=vcfs_or_lookup,
+                use_cache=use_cache,
+                min_maf=min_maf,
+                include_missing_mafs=include_missing_mafs,
+            )
```

Comment:
**issue** While the parameter may be a list of VCFs or a `VariantLookup`, the attribute may only be a `VariantLookup`, and we should name it and type it accordingly. 

Also, what happens if the conditions in these two branches aren't met?

```suggestion
        variant_lookup: Optional[VariantLookup]
        if vcfs_or_lookup is None or isinstance(vcfs_or_lookup, VariantLookup):
            variant_lookup = vcfs_or_lookup
        elif isinstance(vcfs_or_lookup, list) and all(
            isinstance(path, Path) for path in vcfs_or_lookup
        ):  # if provided path, create appropriate `VariantLookup`
            variant_lookup = VariantLookup.build(
                vcf_paths=vcfs_or_lookup,
                use_cache=use_cache,
                min_maf=min_maf,
                include_missing_mafs=include_missing_mafs,
            )
        
        self.variant_lookup = variant_lookup
```